gateway/errors.go

```diff
@@ -20,12 +28,8 @@ import (
 // Addition bool argument indicates whether method (http.ResponseWriter.WriteHeader) was called or not.
 type ProtoStreamErrorHandlerFunc func(context.Context, bool, *runtime.ServeMux, runtime.Marshaler, http.ResponseWriter, *http.Request, error)
 
-// RestError represents an error in accordance with REST API Syntax Specification.
-// See: https://github.com/infobloxopen/atlas-app-toolkit#errors
-type RestError struct {
-	Status  *RestStatus   `json:"error,omitempty"`
-	Details []interface{} `json:"details,omitempty"`
-	Fields  interface{}   `json:"fields,omitempty"`
+type RestErrs struct {
+	Error []map[string]interface{} `json:"error,omitempty"`
```

Comment:
Ah, that's exactly what the `[]map[string]interface{}` does become though. The case for an `interface{}` instead of `string` would be if the user wants to include a numeric value in an error (although it also allows for arbitrary types, including nested messages). Something like:
```
{
    "error": [{
        "message": "maximum value exceeded",
        "value": 100,
        "max": 10
    }...]
}
```
Perhaps it's a bit confusing because I called it `RestErrs`, but this is the type of message returned when a call fails, and only errors are returned.

Reply:
Got it. I missed the part that error may have more fields than just `message`.